pydantic/_internal/_generate_schema.py

```diff
@@ -610,50 +791,30 @@ def _generate_schema(self, obj: Any) -> core_schema.CoreSchema:  # noqa: C901
 
         if _typing_extra.origin_is_union(origin):
             return self._union_schema(obj)
-        elif issubclass(origin, typing.Tuple):  # type: ignore[arg-type]
+        elif origin in TUPLE_TYPES:
```

Comment:
now that we are doing so much less `issubclass` and so much more `origin in ...`, it might make sense to refactor this into a single dict lookup rather than a bunch of if-elses. Doesn't need to happen in this PR, but I wanted to point it out